### PR TITLE
[ML] Fix cause of multimodal prior error

### LINKED ISSUE #274: [ML] Error - MERGE: Sum mode samples = 0, total samples = 4.43521
Found in build

```
./platform.orig/darwin-x86_64/bin/autodetect --version
Model State Version 34
Quantile State Version 3
autodetect (64 bit): Version 7.0.0-alpha1-SNAPSHOT (Build 5aafd3b0eb2dc3) Copyright (c) 2018 Elasticsearch BV
```

Using data-set Iowa-liquor (https://data.iowa.gov/Economy/Iowa-Liquor-Sales/m3tr-qhgy)

with job config

```
  "analysis_config": {
    "bucket_span": "7d",
    "detectors": [
      {
        "detector_description": "sum(SaleDollars) partitionfield=\"CategoryName.keyword\"",
        "function": "sum",
        "field_name": "SaleDollars",
        "partition_field_name": "CategoryName.keyword",
        "detector_index": 0
      }
    ],
    "influencers": [
      "CategoryName.keyword",
      "City.keyword"
    ]
  }
```

Errors seen in log file

```
[2018-10-23T12:39:47,939][ERROR][o.e.x.m.j.p.l.CppLogMessageHandler] [Edwards-MacBook-Pro.local] [iowa-liquor-sum-dollars-by-category_name7d] [autodetect/57567] [CMultimodalPrior.cc@596] MERGE: Sum mode samples = 0, total samples = 4.43521
[2018-10-23T12:39:52,020][ERROR][o.e.x.m.j.p.l.CppLogMessageHandler] [Edwards-MacBook-Pro.local] [iowa-liquor-sum-dollars-by-category_name7d] [autodetect/57567] [CMultimodalPrior.cc@596] SPLIT: Sum mode samples = 8.98052, total samples = 13.4024
[2018-10-23T12:39:52,020][ERROR][o.e.x.m.j.p.l.CppLogMessageHandler] [Edwards-MacBook-Pro.local] [iowa-liquor-sum-dollars-by-category_name7d] [autodetect/57567] [CMultimodalPrior.cc@696] # samples = 8.98052, # modes = 2, pLeft = 0.148765, pRight = 0.851235
[2018-10-23T12:39:52,865][ERROR][o.e.x.m.j.p.l.CppLogMessageHandler] [Edwards-MacBook-Pro.local] [iowa-liquor-sum-dollars-by-category_name7d] [autodetect/57567] [CMultimodalPrior.cc@596] MERGE: Sum mode samples = 0, total samples = 4.65186
```

These errors have not been observed in jobs with bucket length other than 7 days

### PR DESCRIPTION
**Pull request by edsavage**
Removed check for 'non informative' modes when sampling marginal
likelihood in COneOfNPrior. The check is unneeded as the modes all
return some form of sample in any case.

fixes #274

### CODE REVIEW

File: lib/maths/CGammaRateConjugate.cc [hunk 1]
```diff
@@ -1431,7 +1431,10 @@ void CGammaRateConjugate::print(const std::string& indent, std::string& result)
 
     result += core_t::LINE_ENDING + indent + "gamma ";
     if (this->isNonInformative()) {
-        result += "non-informative";
+        result +=
+            "non-informative: # sample moments = " +
+            core::CStringUtils::typeToStringPretty(CBasicStatistics::count(m_SampleMoments)) +
```
Comment (tveasey) [inline lib/maths/CGammaRateConjugate.cc]: nit: maybe "# sample moments" -> "sample count"

**Comment by edsavage** [comment]
Ideally I'd like to get this fix into 6.5 if it's not too late to do so. The effect on results is minimal..

File: lib/maths/unittest/CMultimodalPriorTest.cc [hunk 2]
```diff
@@ -1113,15 +1113,13 @@ void CMultimodalPriorTest::testSampleMarginalLikelihood() {
...
     for (std::size_t i = 0u; i < 3u; ++i) {
         LOG_DEBUG(<< "sample = " << samples[i]);
 
         sampleMoments.add(samples[i]);
         filter.addSamples(TDouble1Vec(1, samples[i]));
-
-        sampled.clear();
-        filter.sampleMarginalLikelihood(10, sampled);
-        CPPUNIT_ASSERT_EQUAL(std::size_t(0), sampled.size());
+        CPPUNIT_ASSERT_EQUAL(std::string("\nmultimodal non-informative"), filter.print());
```
Comment (tveasey) [inline lib/maths/unittest/CMultimodalPriorTest.cc]: I would have this check that the moments of `sampled` are similar to those of the samples added so far, you can check against  `sampleMoments` for this.

Comment (edsavage) [reply lib/maths/unittest/CMultimodalPriorTest.cc]: Added checks for mean and skewness - the variance differs wildly.

**Pull request merged**